src/Uno.Extensions.Configuration/Reloader.cs

```diff
@@ -32,10 +35,35 @@ public async Task ReloadAllFileConfigurationProviders(string? configFile = defau
 				}
 				else
 				{
-					var contents = File.ReadAllText(info.PhysicalPath);
-					if (Logger.IsEnabled(LogLevel.Debug)) Logger.LogDebugMessage($@"Contents '{contents}'");
-					if (Logger.IsEnabled(LogLevel.Debug)) Logger.LogDebugMessage($@"Loading from full path '{info.PhysicalPath}'");
-					fp.Load();
+					if (Logger.IsEnabled(LogLevel.Debug))
+					{
+						var contents = File.ReadAllText(info.PhysicalPath);
+						Logger.LogDebugMessage($@"Contents '{contents}'");
+						Logger.LogDebugMessage($@"Loading from full path '{info.PhysicalPath}'");
+					}
+					var read = false;
+					var attempt = 0;
+					while (!read && attempt++ < MaxReadRetries)
+					{
+						await ReadWriteLock.WaitAsync();
+						try
+						{
+							fp.Load();
+						}
+						catch (IOException)
+						{
+							// Only retry on IOExceptions eg sharing violation
+							read = false;
+						}
+						finally
+						{
+							ReadWriteLock.Release();
+						}
+						if (!read)
+						{
+							await Task.Yield();
+						}
+					}
```

Comment:
This may cause a live lock (two or more readers interleaving) with high CPU usage as a result. You may want to use an actual `Task.Delay(50)` instead to introduce entropy.